src/GridFS/Bucket.php

```diff
@@ -583,16 +583,9 @@ public function rename(mixed $id, string $newFilename): void
             return;
         }
 
-        /* If the update resulted in no modification, it's possible that the
-         * file did not exist, in which case we must raise an error. Checking
-         * the write result's matched count will be most efficient, but fall
-         * back to a findOne operation if necessary (i.e. legacy writes).
-         */
-        $found = $updateResult->getMatchedCount() !== null
-            ? $updateResult->getMatchedCount() === 1
-            : $this->collectionWrapper->findFileById($id) !== null;
-
-        if (! $found) {
+        // If the update resulted in no modification, it's possible that the
+        // file did not exist, in which case we must raise an error.
+        if ($updateResult->getMatchedCount() !== 1) {
```

Comment:
I'm not sure we should consider using unacknowledged writes a valid use case, but in case we do there's a functional change here. If we want to preserve the option to use unacknowledged writes with GridFS, we should check `isAcknowledged` instead of relying on the `null` value from `getMatchedCount`:

```suggestion
        /* If the update resulted in no modification, it's possible that the
         * file did not exist, in which case we must raise an error. Checking
         * the write result's matched count will be most efficient, but fall
         * back to a findOne operation if necessary (i.e. legacy writes).
         */
        $found = $updateResult->isAcknowledged()
            ? $updateResult->getMatchedCount() === 1
            : $this->collectionWrapper->findFileById($id) !== null;

        if (! $found) {
```

Reply:
In that case, that's a fix to backport to v1.x, as it already throws an exception when there is an unacknowledged operation.

Reply:
In that case, let's defer to a separate PR, as this does PR not change existing behaviour.

Reply:
> let's defer to a separate PR

I assume a ticket needs to be created for this. Please cross-reference it here after doing so.

Reply:
I just realized that this second `if ($updateResult->getMatchedCount() !== 1)` is always true after `if ($updateResult->getModifiedCount() === 1) return;`.

Do you know what are the "legacy writes" that are mentioned in the comment?

Reply:
Legacy opcodes (e.g. `OP_UPDATE`), which were [removed in MongoDB 6.0](https://www.mongodb.com/docs/manual/release-notes/6.0-compatibility/#legacy-opcodes-removed). At this point, drivers _may_ still use them for unacknowledged writes but all acknowledged writes are guaranteed to use `OP_MSG` and the corresponding write commands (e.g. [update](https://www.mongodb.com/docs/manual/reference/command/update/)).